File: chatmessageview/src/main/java/com/github/bassaer/chatmessageview/utils/IMessageStatusIconFormatter.java
package com.github.bassaer.chatmessageview.utils;

import android.graphics.drawable.Drawable;

/**
 * Interface for status icon format
 * Created by nakayama on 2017/02/16.
 */
public interface IMessageStatusIconFormatter {
    /**
     * Return icon depend on message status and sender
     * @param status message status
     * @param isRightMessage Whether sender is right or not
     * @return status icon image
     */
    Drawable getStatusIcon(int status, boolean isRightMessage);
}

File: chatmessageview/src/main/kotlin/com/github/bassaer/chatmessageview/util/IMessageStatusIconFormatter.kt
package com.github.bassaer.chatmessageview.util

import android.graphics.drawable.Drawable

/**
 * Interface for status icon format
 * Created by nakayama on 2017/02/16.
 */
interface IMessageStatusIconFormatter {
    /**
     * Return icon depend on message status and sender
     * @param status message status
     * @param isRightMessage Whether sender is right or not
     * @return status icon image
     */
    fun getStatusIcon(status: Int, isRightMessage: Boolean): Drawable
}
